lib/tools/api_python_module.py

```diff
@@ -0,0 +1,191 @@
+        """
+        params = {}
+        params['filename'] = title
+        params['file_content'] = contents
+        path = self.get_path(assignment_id, group_id) + 'test_results.xml'
+        return self.submit_request(params, path, 'POST')
+
+    def update_marks_single_group(self, criteria_mark_map,
+                                  assignment_id, group_id):
+        """ (ApiInterface, dict, int, int) -> list of str
+        Update the marks of a single group. 
+        Only the marks specified in criteria_mark_map will be changed.
+        To set a mark to unmarked, use 'nil' as it's value.
+        Otherwise, marks must have valid numeric types (floats or ints).
+        Criteria are specified by their title. Titles must be formatted
+        exactly as they appear in the MarkUs GUI, punctuation included.
+
+        Keyword arguments:
+        criteria_mark_map --  maps criteria to the desired grade.
+        assignment_id -- the assignment's id
+        group_id      -- the id of the group whose marks we are updating
+        """
+        params = criteria_mark_map
+        path = self.get_path(assignment_id, group_id) + 'update_marks'
+        return self.submit_request(params, path, 'PUT')
+
+    def submit_request(self, params, path, request_type):
+        """ (ApiInterface, dict, str, str) -> list of str
+        Perform the HTTP/HTTPS request. Return a list 
+        containing the response's status, reason, and content.
+
+        Keyword arguments:
+        params       -- contains the parameters of the request
+        path         -- route to the resource we are targetting
+        request_type -- the desired HTTP method (usually 'GET' or 'POST')
+        """
+        auth_header = 'MarkUsAuth %s' % self.api_key
+        headers = { 'Authorization': auth_header,
+                    'Content-type': 'application/x-www-form-urlencoded' }
+        if request_type == 'GET': # we only want this for GET requests
+            headers['Accept'] = 'text/plain'
+        if params != None:
+            params = urlencode(params)
+        try:
+            resp = None; conn = None
+            if self.protocol == 'http':
+                conn = http.client.HTTPConnection(self.parsed_url.netloc)
+            elif self.protocol == 'https':
+                conn = http.client.HTTPSConnection(self.parsed_url.netloc)
+            else:
+                print('Panic! Neither http nor https URL.')
+                sys.exit(1)
+            conn.request(request_type,
+                         (self.parsed_url.path + path),
+                         params,
+                         headers)
+            resp = conn.getresponse()
+            lst = [resp.status, resp.reason, resp.read()]
+            for x in lst:
+                print(type(x))
+            conn.close()
+            return lst
+        except http.client.HTTPException as e: # Catch HTTP errors
+            print(str(e), file=sys.stderr)
+            sys.exit(1)
```

Comment:
I'm using sys.exit(1) because that's what the person who wrote api_helper.py used, and I'm not sure what the better alternative is.